pytorch_lightning/utilities/model_summary.py

```diff
@@ -261,16 +261,10 @@ def _forward_example_input(self) -> None:
         input_ = model.example_input_array
         input_ = model._apply_batch_transfer_handler(input_)
 
-        if (
-            trainer is not None
-            and trainer.amp_backend == AMPType.NATIVE
-            and trainer._device_type != _AcceleratorType.TPU
-        ):
-            model.forward = torch.cuda.amp.autocast()(model.forward)
-
         mode = model.training
         model.eval()
-        with torch.no_grad():
+
+        with torch.no_grad(), (trainer.precision_plugin.forward_context() if trainer else contextlib.nullcontext()):
```

Comment:
I think this syntax won't work with older python versions (3.6 and 3.7)

Reply:
yeah looks like just py3.6.
Will wait for this: https://github.com/PyTorchLightning/pytorch-lightning/pull/11117

Reply:
done!